src/aleph/schemas/api/messages.py

```diff
@@ -97,19 +108,30 @@ class StoreMessage(
 }
 
 
-def format_message(message: Any) -> AlephMessage:
-    message_cls = MESSAGE_CLS_DICT[message.type]
-    return message_cls.from_orm(message)
+AlephMessage = Annotated[
+    Union[
+        AggregateMessage,
+        ForgetMessage,
+        InstanceMessage,
+        PostMessage,
+        ProgramMessage,
+        StoreMessage,
+    ],
+    Field(discriminator="type"),
+]
 
 
-AlephMessage = Union[
-    AggregateMessage,
-    ForgetMessage,
-    InstanceMessage,
-    PostMessage,
-    ProgramMessage,
-    StoreMessage,
-]
+def format_message(message: MessageDb) -> AlephMessage:
+    message_type = message.type
+
+    message_cls = MESSAGE_CLS_DICT[message_type]
+    return message_cls.from_orm(message)  # type: ignore[return-value]
```

Comment:
What is the issue here ?

Reply:
mypy does not infer the type of `message_cls` correctly. Says it's an instance of `BaseMessage`, which is correct, but does not match the `AlephMessage` union type. Basically unhappy because it can't figure out that `AlephMessage` is a union of all the possible child classes of `BaseMessage`.